src/nvofapi_d3d12.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "nvofapi/nvofapi_entrypoints.h"
+#include "nvofapi/nvofapi_image.h"
+#include "d3d12/nvofapi_d3d12_instance.h"
+
+#include "util/util_env.h"
+#include "util/util_log.h"
+#include "util/util_string.h"
+#include "util/util_statuscode.h"
+#include "../version.h"
+
+using namespace dxvk;
+extern "C" {
+
+    // D3D12 entrypoints
+    NV_OF_STATUS NVOFAPI CreateOpticalFlowD3D12(ID3D12Device* pD3D12Device, NvOFHandle* hOFInstance) {
+        constexpr auto n = __func__;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(pD3D12Device), log::fmt::ptr(hOFInstance));
+
+        nvofapi::NvOFInstanceD3D12* nvOF = nullptr;
+        try {
+            nvOF = new nvofapi::NvOFInstanceD3D12(pD3D12Device);
+        } catch (std::exception const& e) {
+            log::info(str::format("CreateOpticalFlowD3D12 exception, %s", e.what()));
+        }
+
+        if (!nvOF) {
```

Comment:
I'm not sure that I agree given that if the constructor throws an exception (either OOM, or otherwise) the assignment shouldn't complete and the value should remain `nullptr`. Either way I re-worked this a bit into a way that I think will mesh a bit better.